perun/check/local_statistics.py

```diff
@@ -0,0 +1,355 @@
+        y_pts = reshape_array(y_pts)
+        x_pts = reshape_array(x_pts)
+        # obtain the first and last point from the sub-intervals - edges of the intervals
+        x_edges = np.delete(x_pts, range(1, x_pts[0].size - 1), 1)
+        # replace the right edge of the last sub-interval with the saved maximum of whole interval:
+        # - (10, NaN) -> (10, 10)
+        x_edges[np.isnan(np.delete(x_pts, range(1, x_pts[0].size - 1), 1))] = max_x
+    # compute the statistical metrics on the whole interval of x-coordinates
+    else:
+        # obtain the edges of the whole interval
+        x_edges = np.delete(x_pts, range(1, x_pts.size - 1)).reshape(1, -1)
+
+    # compute the all statistical metrics on the specified intervals
+    return {
+        # integral
+        'int': np.atleast_1d(integrate.simps(y_pts, x_pts, axis=axis)),
+        # average/mean
+        'avg': np.atleast_1d(np.nanmean(y_pts, axis=axis)),
+        # median/2.percentile
+        'med': np.atleast_1d(np.nanmedian(y_pts, axis=axis)),
+        # maximum
+        'max': np.atleast_1d(np.nanmax(y_pts, axis=axis)),
+        # minimum
+        'min': np.atleast_1d(np.nanmin(y_pts, axis=axis)),
+        # summary value
+        'sum': np.atleast_1d(np.nansum(y_pts, axis=axis)),
+        # 1.percentile
+        'per_q1': np.atleast_1d(np.nanpercentile(y_pts, 25, axis=axis)),
+        # 3.percentile
+        'per_q2': np.atleast_1d(np.nanpercentile(y_pts, 75, axis=axis)),
+    }, x_edges
+
+
+def classify_stats_diff(base_stats, targ_stats):
+    """
+    The method performs the classification of computed statistical metrics.
+
+    A method analyses the computed statistical metrics in the given lists. Between
+    the pair of metrics is computed the relative error and subsequently according to
+    its value is determined the change state. The resulting change on the individual
+    sub-interval is determined by the final score of change, that is computed during
+    the iteration over all metrics. Finally, a method determines the change states
+    and computes the average relative error.
+
+    :param dict base_stats: contains the metrics on relevant sub-intervals from baseline model
+    :param dict targ_stats: contains the metrics on relevant sub-intervals from target model
+    :return tuple: (change states on individual sub-intervals, average relative error)
+    """
+    # create vectorized functions which take a np.arrays as inputs and perform actions over it
+    compare_diffs = np.vectorize(compare_diff_values)
+    classify_change = np.vectorize(check_helpers.classify_change)
+    # initialize np.arrays about the size of sub-intervals from given stats
+    change_score = np.full_like(base_stats.get(list(base_stats.keys())[0]), 0)
+    rel_error = np.full_like(base_stats.get(list(targ_stats.keys())[0]), 0)
+    # iteration over each metric in the given stats
+    for base_stat in base_stats.items():
```

Comment:
Because `baseline_stats` and `target_stats` are dictionaries and I want to obtain the items with the same keys from both dictionaries. Therefore I iterate over the `baseline_stats` and according to the key of the current item I get the relevant item from the `target_stats`.

Reply:
Aw yeah, my bad, I made some hasty assumptions :). Then I propose to extract this to `for (base_stat_key, base_stat_value)` it should not make anything too long and will be better readable.